File: problem=010/Solution.java
import java.util.Arrays;

final class Solution {

	private static void sieve(boolean[] primes) {
		Arrays.fill(primes, true);
		for (int i = 2; i < primes.length; i++) {
			if (primes[i]) {
				for (int j = (i << 1); j < primes.length; j += i) {
					primes[j] = false;
				}
			}
		}
	}

	private static long solve(int limit) {
		final boolean[] primes = new boolean[limit + 1];
		sieve(primes);
		long sum = 0;
		for (int i = 2; i < primes.length; i++) {
			if (primes[i]) {
				sum += i;
			}
		}
		return sum;
	}

	public static void main(String[] args) throws Exception {
		System.out.println(solve(10));
		System.out.println(solve((int) 2e6));
	}
}

// If this solution helped you, please star the repo!

File: problem=010/Solution.js
const sieve = (primes = []) => {
	primes.fill(true);
	for (let i = 2; i < primes.length; i++) {
		if (primes[i]) {
			for (let j = i << 1; j < primes.length; j += i) {
				primes[j] = false;
			}
		}
	}
};

const solve = limit => {
	const primes = new Array(limit + 1);
	sieve(primes);
	let sum = 0;
	for (let i = 2; i < primes.length; i++) {
		if (primes[i]) {
			sum += i;
		}
	}
	return sum;
};

console.log(solve(10));
console.log(solve(2e6));

// If this solution helped you, please star the repo!
